lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -1126,38 +1128,32 @@ void FIRRTLDialect::registerTypes() {
 // field element and return the total bit width of the aggregate type. This
 // returns None, if any of the bundle fields is a flip type, or ground type with
 // unknown bit width.
-llvm::Optional<int64_t> firrtl::getBitWidth(FIRRTLBaseType type,
-                                            bool ignoreFlip) {
-  std::function<llvm::Optional<int64_t>(FIRRTLBaseType)> getWidth =
-      [&](FIRRTLBaseType type) -> llvm::Optional<int64_t> {
-    return TypeSwitch<FIRRTLBaseType, llvm::Optional<int64_t>>(type)
-        .Case<BundleType>([&](BundleType bundle) {
+std::optional<int64_t> firrtl::getBitWidth(FIRRTLBaseType type,
+                                           bool ignoreFlip) {
+  std::function<std::optional<int64_t>(FIRRTLBaseType)> getWidth =
+      [&](FIRRTLBaseType type) -> std::optional<int64_t> {
+    return TypeSwitch<FIRRTLBaseType, std::optional<int64_t>>(type)
+        .Case<BundleType>([&](BundleType bundle) -> std::optional<int64_t> {
           int64_t width = 0;
           for (auto &elt : bundle) {
             if (elt.isFlip && !ignoreFlip)
-              return llvm::Optional<int64_t>(std::nullopt);
+              return std::nullopt;
             auto w = getBitWidth(elt.type);
             if (!w.has_value())
-              return llvm::Optional<int64_t>(std::nullopt);
+              return std::nullopt;
             width += *w;
           }
-          return llvm::Optional<int64_t>(width);
+          return width;
         })
-        .Case<FVectorType>([&](auto vector) {
+        .Case<FVectorType>([&](auto vector) -> std::optional<int64_t> {
           auto w = getBitWidth(vector.getElementType());
           if (!w.has_value())
-            return llvm::Optional<int64_t>(std::nullopt);
-          return llvm::Optional<int64_t>(*w * vector.getNumElements());
-        })
-        .Case<IntType>([&](IntType iType) {
-          auto retval = iType.getWidth();
-          if (retval)
-            return llvm::Optional<int64_t>(*retval);
-          else
-            return llvm::Optional<int64_t>(std::nullopt);
+            return std::nullopt;
+          return *w * vector.getNumElements();
         })
+        .Case<IntType>([&](IntType iType) { return iType.getWidth(); })
```

Comment:
Nice that this simplifies.

Reply:
Yeah! It does perform an extension from the int32_t optional in `getWidth()` which didn't work before with `llvm::Optional`? (`Optional<int32_t>` -> `Optional<int64_t>`).